### PR TITLE
luarocks: update to `3.9.1`

### PR DESCRIPTION
**Pull request by UrNightmaree**
Bump Luarocks package to 3.9.1 and also changes Lua dependency from 5.3 to 5.4

### CODE REVIEW

File: packages/luarocks/build.sh [hunk 1]
```diff
@@ -1,37 +1,37 @@
 TERMUX_PKG_HOMEPAGE=https://luarocks.org/
 TERMUX_PKG_DESCRIPTION="Deployment and management system for Lua modules"
-TERMUX_PKG_LICENSE="BSD"
```
Comment (thunder-coding) [inline packages/luarocks/build.sh:3]: Ah that license. Could have caused huge havoc if we were doing the same with some non-OSS stuff. Thanks for rectifying

Comment (UrNightmaree) [reply packages/luarocks/build.sh:3]: Do I need to do something with it?

Comment (thunder-coding) [reply packages/luarocks/build.sh:3]: Nothing, I was just thanking you for fixing it :)

Comment (UrNightmaree) [reply packages/luarocks/build.sh:3]: Oh, Ok 👌

**Pull request closed without merging**